src/main/java/org/junit/rules/SortRules.java

```diff
@@ -0,0 +1,86 @@
+package org.junit.rules;
+
+import static org.junit.rules.RuleChain.emptyRuleChain;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+/*
+ * Stores a list of TestRule-priorities-pairs
+ * and can return a lists of all TestRules
+ * as well as a list containing the autoconstructed RuleChain.
+ */
+public class SortRules {
+    private final List<RulePriorityPair> rules = new ArrayList<RulePriorityPair>();
+
+    /*
+     * Adds a TestRulePriorityPair into the List.
+     * @param rule: the TestRule to be added to SortRules
+     * @param priority: the annotated priority of the TestRule
+     */
+    public void add(TestRule rule, int priority) {
+        rules.add(new RulePriorityPair(rule,priority));
+    }
+
+    /*
+     * Returns the a list of all the stored TestRules in insertion order.
+     * @return A list of TestRules in the order of their insertion.
+     */
+    public List<TestRule> listWithAllRules() {
+        List<TestRule> result = new ArrayList<TestRule>(rules.size());
+        for (RulePriorityPair rule : rules) {
+            result.add(rule.testRule);
+        }
+        return result;
+    }
+
+    /*
+     * Construct a list of all TestRules with negative priorities
+     * plus a autoconstructed RuleChain containing all TestRules with non-negative priorities.
+     * The TestRules of the RuleChain are sorted in such a way
+     * that the TestRule with the highest priority becomes the most outer TestRule.
+     */
+    public List<TestRule> listWithAutoconstructedRuleChain() {
```

Comment:
This is a bit of an awkward name. How about we make this method package-scope, and add a public method named `toStatement()` with class this method and then does the work that is currently done in `withTestRules(FrameworkMethod, List<TestRule>, Statement)`?